src/function/scalar/string/mismatches.cpp

```diff
@@ -0,0 +1,47 @@
+#include "duckdb/function/scalar/string_functions.hpp"
+#include "duckdb/common/vector_operations/vector_operations.hpp"
+
+#include <ctype.h>
+#include <algorithm>
+
+namespace duckdb {
+
+static int64_t MismatchesScalarFunction(Vector &result, const string_t str, string_t tgt) {
+	int64_t num_characters_str = LengthFun::Length<string_t, int64_t>(str);
+	int64_t num_characters_tgt = LengthFun::Length<string_t, int64_t>(tgt);
+	if (num_characters_str != num_characters_tgt) {
+		return -1;  // Expected error code: both strings must be of equal length
+	}
+	if (num_characters_str < 1) {
+		return -1; // too shorts
+	}
+
+	int64_t mismatches = 0;
+
+	for (int64_t idx = 0; idx < (int64_t)str.GetSize(); ++idx) {
+		if (str.GetString()[idx] != tgt.GetString()[idx]) {
```

Comment:
Please avoid using `GetString()`, especially in a loop such as this: two new strings will be constructed for every character in the string. Instead, use `GetDataUnsafe()` and store the resulting pointer in a temporary variable, e.g.:


```cpp
auto str_data = str.GetDataUnsafe();
auto tgt_data = tgt.GetDataUnsafe();
for (...) {
    if (str_data[idx] != tgt_data[idx]) { ... }
}
```

Reply:
Thanks!